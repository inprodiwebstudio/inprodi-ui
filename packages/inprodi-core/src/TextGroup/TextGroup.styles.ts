import { createStyles, DefaultMantineColor } from "@mantine/core";

const styles = createStyles((theme, { color } : { color ?: DefaultMantineColor } ) => {
	return {
		root : {
			"& svg" : {
				color : color
					?  theme.fn.variant({ variant : "filled", color }).background
					: theme.colorScheme === "dark" ? theme.colors.dark[3]: theme.colors.gray[6],
			},
		},
	};
});

export default styles;
