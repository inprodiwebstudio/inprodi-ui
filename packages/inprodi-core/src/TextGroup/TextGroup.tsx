import { Text }                       from "@mantine/core";
import { Flex, TextProps, FlexProps } from "@mantine/core";
import useStyles                      from "./TextGroup.styles";

export interface TextGroupProps extends TextProps {
    leftIcon ?: React.ReactNode;
    rightIcon ?: React.ReactNode;
	wrapperProps ?: FlexProps
}

export const TextGroup = ({
	color,
	children,
	leftIcon,
	rightIcon,
	wrapperProps,
	...rest
} : TextGroupProps) => {
	const { classes } = useStyles({ color });
	return (
		<Flex
			data-testid="text-group-flex"
			gap={20}
			align="center"
			className={classes.root}
			{...wrapperProps}
		>
			{leftIcon && leftIcon}
			<Text
				data-testid="text-group-text"
				color={color}
				{...rest}
			>
				{children}
			</Text>
			{rightIcon && rightIcon}
		</Flex>
	);
};

export default TextGroup;
