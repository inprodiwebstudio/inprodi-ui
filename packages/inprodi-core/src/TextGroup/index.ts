export { TextGroup } from "./TextGroup";
export type { TextGroupProps } from "./TextGroup";
