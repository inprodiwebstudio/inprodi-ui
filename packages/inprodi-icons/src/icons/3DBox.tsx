import * as React from "react";
import { SVGProps } from "react";
const Svg3DBox = (props: SVGProps<SVGSVGElement>) => (
  <svg
    width="1em"
    height="1em"
    viewBox="0 0 14 14"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <path
      d="M6.646 0a3.32 3.32 0 0 0-1.514.365h-.001L1.043 2.47C.403 2.8 0 3.46 0 4.18v5.64c0 .719.403 1.38 1.043 1.71l4.088 2.104v.001c.95.487 2.079.487 3.029 0h.001l4.088-2.105c.64-.33 1.042-.991 1.042-1.71V4.18c0-.719-.402-1.38-1.042-1.71L8.16.366V.365A3.323 3.323 0 0 0 6.646 0Zm0 1.05c.355 0 .711.083 1.036.25l4.086 2.103a.87.87 0 0 1 .474.777v5.64a.872.872 0 0 1-.474.777L7.682 12.7c-.65.334-1.42.334-2.071.001H5.61l-4.086-2.104a.871.871 0 0 1-.475-.778V4.18c0-.329.182-.627.475-.777L5.61 1.299a2.27 2.27 0 0 1 1.036-.25ZM2.263 3.673a.525.525 0 0 0-.231.993l3.09 1.605c.316.165.656.254 1 .31v5.14a.525.525 0 1 0 1.049 0v-5.14a3.271 3.271 0 0 0 1-.31l3.09-1.604a.525.525 0 1 0-.484-.932L7.687 5.34a2.256 2.256 0 0 1-1.051.252 2.256 2.256 0 0 1-1.031-.252l-3.09-1.605a.525.525 0 0 0-.252-.063Zm8.747 2.969a.525.525 0 0 0-.516.532v1.897c0 .007-.002.011-.008.014h-.001l-1.456.75a.525.525 0 1 0 .481.933l1.455-.75c.354-.182.578-.55.578-.947V7.175a.525.525 0 0 0-.533-.532Z"
      fill="currentColor"
    />
  </svg>
);
export default Svg3DBox;
