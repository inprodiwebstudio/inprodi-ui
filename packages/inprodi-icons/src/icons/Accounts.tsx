import * as React from "react";
import { SVGProps } from "react";
const SvgAccounts = (props: SVGProps<SVGSVGElement>) => (
  <svg
    width="1em"
    height="1em"
    viewBox="0 0 14 15"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <path
      d="M6.014.6c-.403 0-.805.131-1.137.393L.191 4.685a.501.501 0 0 0 .31.895h11.025a.501.501 0 0 0 .31-.895L7.151.993A1.835 1.835 0 0 0 6.014.6Zm0 1c.182 0 .364.06.517.18l3.55 2.797H1.946l3.55-2.796a.834.834 0 0 1 .518-.181Zm0 .973a.668.668 0 1 0 0 1.336.668.668 0 0 0 0-1.336ZM1.169 6.248v4.383C.501 10.784 0 11.381 0 12.094c0 .829.675 1.504 1.503 1.504h3.508v-1.003H1.503a.5.5 0 1 1 0-1.002h3.508v-1.002h-.668V6.248H3.341v4.343h-1.17V6.248H1.17Zm4.344 0V8.8a1.13 1.13 0 0 1 .668-.214h.334V6.248H5.513Zm2.171 0v2.338h1.002V6.248H7.684Zm2.172 0v2.338h1.002V6.248H9.856ZM6.515 9.255a.843.843 0 0 0-.835.835v3.675c0 .455.38.835.835.835h6.013c.456 0 .836-.38.836-.835V10.09a.843.843 0 0 0-.836-.835H6.515Zm.167 1.002h5.68v.668h-5.68v-.668Zm0 1.67h5.68v1.67h-5.68v-1.67Z"
      fill="currentColor"
    />
  </svg>
);
export default SvgAccounts;
