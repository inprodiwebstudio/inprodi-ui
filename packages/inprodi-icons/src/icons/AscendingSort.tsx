import * as React from "react";
import { SVGProps } from "react";
const SvgAscendingSort = (props: SVGProps<SVGSVGElement>) => (
  <svg
    width="1em"
    height="1em"
    viewBox="0 0 14 15"
    fill="none"
    xmlns="http://www.w3.org/2000/svg"
    {...props}
  >
    <path
      d="M10.68.6a.583.583 0 0 0-.573.591v8.304L8.383 7.771a.583.583 0 1 0-.824.824l2.719 2.719a.583.583 0 0 0 .824 0l2.719-2.72a.584.584 0 1 0-.824-.823l-1.724 1.724V1.191A.582.582 0 0 0 10.68.6ZM.59.997a.583.583 0 1 0 0 1.165h1.166a.583.583 0 1 0 0-1.165H.591Zm0 3.107a.583.583 0 1 0 0 1.165h2.72a.583.583 0 1 0 0-1.165H.59Zm0 3.108a.583.583 0 1 0 0 1.165h4.274a.582.582 0 1 0 0-1.165H.59Zm0 3.107a.582.582 0 1 0 0 1.165h5.827a.582.582 0 1 0 0-1.165H.591Zm0 3.107a.584.584 0 0 0-.416.998.583.583 0 0 0 .417.168h7.38a.582.582 0 1 0 0-1.166H.591Z"
      fill="currentColor"
    />
  </svg>
);
export default SvgAscendingSort;
