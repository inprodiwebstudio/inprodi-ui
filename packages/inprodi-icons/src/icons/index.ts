export { default as Svg3DBox } from "./3DBox";
export { default as Accounts } from "./Accounts";
export { default as AscendingSort } from "./AscendingSort";
export { default as Attachment } from "./Attachment";
export { default as Bell } from "./Bell";
export { default as Book } from "./Book";
export { default as Briefcase } from "./Briefcase";
export { default as Broke } from "./Broke";
export { default as Bug } from "./Bug";
export { default as Buildings } from "./Buildings";
export { default as BurgerMenu } from "./BurgerMenu";
export { default as Calendar } from "./Calendar";
export { default as Cart } from "./Cart";
export { default as CashRegister } from "./CashRegister";
export { default as Catalogue } from "./Catalogue";
export { default as Chart } from "./Chart";
export { default as Chat } from "./Chat";
export { default as CheckCircle } from "./CheckCircle";
export { default as Check } from "./Check";
export { default as ChevronDown } from "./ChevronDown";
export { default as ChevronRight } from "./ChevronRight";
export { default as Circles } from "./Circles";
export { default as ClosedEye } from "./ClosedEye";
export { default as Columns } from "./Columns";
export { default as ContactBook } from "./ContactBook";
export { default as Content } from "./Content";
export { default as Contract } from "./Contract";
export { default as DescendingSort } from "./DescendingSort";
export { default as Devices } from "./Devices";
export { default as Diagram } from "./Diagram";
export { default as DiscussionGroup } from "./DiscussionGroup";
export { default as EnterArrow } from "./EnterArrow";
export { default as Exit } from "./Exit";
export { default as FilterSort } from "./FilterSort";
export { default as Filter } from "./Filter";
export { default as Folder } from "./Folder";
export { default as Gears } from "./Gears";
export { default as Grid } from "./Grid";
export { default as Info } from "./Info";
export { default as Invoice } from "./Invoice";
export { default as Key } from "./Key";
export { default as LeftArrow } from "./LeftArrow";
export { default as Link } from "./Link";
export { default as Lock } from "./Lock";
export { default as Mail } from "./Mail";
export { default as MasonryGrid } from "./MasonryGrid";
export { default as Menu } from "./Menu";
export { default as Moon } from "./Moon";
export { default as Note } from "./Note";
export { default as Octahedron } from "./Octahedron";
export { default as OpenedBox } from "./OpenedBox";
export { default as OpenedEye } from "./OpenedEye";
export { default as Paper } from "./Paper";
export { default as Pdf } from "./Pdf";
export { default as Pencil } from "./Pencil";
export { default as Percentage1 } from "./Percentage1";
export { default as Percentage } from "./Percentage";
export { default as Plus } from "./Plus";
export { default as Preferences } from "./Preferences";
export { default as Project } from "./Project";
export { default as Random } from "./Random";
export { default as RecieveMoney } from "./RecieveMoney";
export { default as Rows } from "./Rows";
export { default as Search } from "./Search";
export { default as SendMoney } from "./SendMoney";
export { default as Send } from "./Send";
export { default as ShieldBlock } from "./ShieldBlock";
export { default as ShieldCheck } from "./ShieldCheck";
export { default as ShieldX } from "./ShieldX";
export { default as Smartphone } from "./Smartphone";
export { default as Sun } from "./Sun";
export { default as Support } from "./Support";
export { default as Ticket } from "./Ticket";
export { default as Tools } from "./Tools";
export { default as Trash } from "./Trash";
export { default as UserGroup } from "./UserGroup";
export { default as UserPlus } from "./UserPlus";
export { default as UserProfessional } from "./UserProfessional";
export { default as UserRelation } from "./UserRelation";
export { default as User } from "./User";
export { default as Vote } from "./Vote";
export { default as Warning } from "./Warning";
export { default as X } from "./X";
