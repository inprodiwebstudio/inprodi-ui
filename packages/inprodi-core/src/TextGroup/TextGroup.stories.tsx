import React                             from "react";
import { CirclesOutline }                from "@inprodi/icons";
import { ComponentMeta, ComponentStory } from "@storybook/react";
import { TextGroup, TextGroupProps }     from "./TextGroup";

export default {
	title     : "Core/Typography/TextGroup",
	component : TextGroup,
	argTypes  : {
		size      : { control : "text" },
		color     : { control : "color" },
		rightIcon : { control : "disabled" },
		leftIcon  : { control : "disabled" },
	},
} as ComponentMeta<typeof TextGroup>;

const Template: ComponentStory<typeof TextGroup> = (args : TextGroupProps) => {
	return(
		<TextGroup {...args}>Text Icon</TextGroup>
	);
};

export const LeftIcon = Template.bind({});
LeftIcon.args = {
	leftIcon : <CirclesOutline />,
};
export const RightIcon = Template.bind({});
RightIcon.args = {
	rightIcon : <CirclesOutline />,
};
export const BothIcon = Template.bind({});
BothIcon.args = {
	leftIcon  : <CirclesOutline />,
	rightIcon : <CirclesOutline />,
};
export const CustomColors = Template.bind({});
CustomColors.args = {
	color     : "blue",
	leftIcon  : <CirclesOutline color="#BE4BDB" />,
	rightIcon : <CirclesOutline color="#9775FA" />,
};
export const WrapperStyled = Template.bind({});
WrapperStyled.args = {
	leftIcon     : <CirclesOutline />,
	rightIcon    : <CirclesOutline />,
	wrapperProps : {
		p         : 15,
		gap       : 10,
		justify   : "end",
		bg        : "dark",
		direction : "column",
		align     : "center",
	},
};
