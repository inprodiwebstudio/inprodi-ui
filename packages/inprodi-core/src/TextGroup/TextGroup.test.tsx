import React              from "react";
import { render }         from "@testing-library/react";
import { CirclesOutline } from "@inprodi/icons";
import { TextGroup }      from "./TextGroup";

describe("TextGroup", () => {
	test("snapshot", () => {
		const { container } = render(
			<TextGroup leftIcon={<CirclesOutline />} rightIcon={<CirclesOutline />}>Label</TextGroup>
		);
		expect(container).toMatchSnapshot();
	});
	test("should has all the correct classNames", () => {
		const flexClassName = "flex-className";
		const textClassName = "text-className";
		const { getByTestId } = render(
			<TextGroup className={textClassName} wrapperProps={{ className : flexClassName }}>Label</TextGroup>
		);
		expect(getByTestId("text-group-flex").className.includes(flexClassName)).toBeTruthy();
		expect(getByTestId("text-group-text").className.includes(textClassName)).toBeTruthy();
	});
});
