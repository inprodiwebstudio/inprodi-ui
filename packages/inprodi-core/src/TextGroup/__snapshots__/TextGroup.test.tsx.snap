// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TextGroup snapshot 1`] = `
<div>
  <div
    class="mantine-zyyu8x"
    data-testid="text-group-flex"
  >
    <svg
      fill="none"
      height="16"
      viewBox="0 0 14 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 .774c-3.86 0-7 3.14-7 7s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7Zm0 1.05a5.942 5.942 0 0 1 5.95 5.95A5.942 5.942 0 0 1 7 13.724a5.942 5.942 0 0 1-5.95-5.95A5.942 5.942 0 0 1 7 1.824Zm0 2.8a3.158 3.158 0 0 0-3.15 3.15A3.158 3.158 0 0 0 7 10.924a3.158 3.158 0 0 0 3.15-3.15A3.158 3.158 0 0 0 7 4.624Zm0 1.05c1.166 0 2.1.934 2.1 2.1s-.934 2.1-2.1 2.1-2.1-.934-2.1-2.1.934-2.1 2.1-2.1Z"
        fill="currentColor"
      />
    </svg>
    <div
      class="mantine-Text-root mantine-1pbxw0k"
      data-testid="text-group-text"
    >
      Label
    </div>
    <svg
      fill="none"
      height="16"
      viewBox="0 0 14 16"
      width="16"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M7 .774c-3.86 0-7 3.14-7 7s3.14 7 7 7 7-3.14 7-7-3.14-7-7-7Zm0 1.05a5.942 5.942 0 0 1 5.95 5.95A5.942 5.942 0 0 1 7 13.724a5.942 5.942 0 0 1-5.95-5.95A5.942 5.942 0 0 1 7 1.824Zm0 2.8a3.158 3.158 0 0 0-3.15 3.15A3.158 3.158 0 0 0 7 10.924a3.158 3.158 0 0 0 3.15-3.15A3.158 3.158 0 0 0 7 4.624Zm0 1.05c1.166 0 2.1.934 2.1 2.1s-.934 2.1-2.1 2.1-2.1-.934-2.1-2.1.934-2.1 2.1-2.1Z"
        fill="currentColor"
      />
    </svg>
  </div>
</div>
`;
